vcf.c

```diff
@@ -3016,8 +3022,16 @@ int bcf_unpack(bcf1_t *b, int which)
 int vcf_format(const bcf_hdr_t *h, const bcf1_t *v, kstring_t *s)
 {
     int i;
+    int32_t max_dt_id = h->n[BCF_DT_ID];
```

Comment:
In `bcf_record_check`, when initializing `max_id`, you checked first the header pointer for NULL value. Is there no need for that here?

Reply:
If `bcf_record_check()` gets called via an iterator read there's no header available so it is passed in as NULL and the function needs to deal with that.  `vcf_format()` won't work at all without a header so it's a coding error if `h == NULL`.  I guess you could argue that it should either return an error code or abort in that case (and the same would be true for `v` and `s`).